pyomo/solvers/tests/checks/test_gurobi_direct.py

```diff
@@ -0,0 +1,257 @@
+    # cases. Expect an error to indicate pyomo tried to set the parameter.
+
+    def test_param_changes_1(self):
+        # Default env: parameters set on model at solve time
+        with SolverFactory("gurobi_direct", options={"Method": -100}) as opt:
+            with self.assertRaisesRegex(gp.GurobiError, "Unable to set"):
+                opt.solve(self.model)
+
+    def test_param_changes_2(self):
+        # Note that this case throws an ApplicationError instead of a
+        # GurobiError since the bad parameter value prevents the environment
+        # from starting
+        # Managed env: parameters set on env at solve time
+        with SolverFactory(
+            "gurobi_direct", options={"Method": -100}, manage_env=True
+        ) as opt:
+            with self.assertRaisesRegex(ApplicationError, "Unable to set"):
+                opt.solve(self.model)
+
+    def test_param_changes_3(self):
+        # Default env: parameters passed to solve()
+        with SolverFactory("gurobi_direct") as opt:
+            with self.assertRaisesRegex(gp.GurobiError, "Unable to set"):
+                opt.solve(self.model, options={"Method": -100})
+
+    def test_param_changes_4(self):
+        # Managed env: parameters passed to solve()
+        with SolverFactory("gurobi_direct", manage_env=True) as opt:
+            with self.assertRaisesRegex(gp.GurobiError, "Unable to set"):
+                opt.solve(self.model, options={"Method": -100})
+
+
+@unittest.skipIf(not gurobipy_available, "gurobipy is not available")
+class GurobiEnvironmentTests(GurobiBase):
+    def test_init_default_env(self):
+        # available() calls with the default env shouldn't need a repeat check
+        with patch("gurobipy.Model") as PatchModel:
+            with SolverFactory("gurobi_direct") as opt:
+                opt.available()
+                opt.available()
+                PatchModel.assert_called_once_with()
+
+    def test_close_global(self):
+        # method releases the license and syncs the flag
+        with patch("gurobipy.Model") as PatchModel, patch("gurobipy.disposeDefaultEnv") as patch_dispose:
+            with SolverFactory("gurobi_direct") as opt:
+                opt.available()
+                opt.available()
+                PatchModel.assert_called_once_with()
+            patch_dispose.assert_not_called()
+
+            opt.close_global()
+            patch_dispose.assert_called_once_with()
+
+        with patch("gurobipy.Model") as PatchModel, patch("gurobipy.disposeDefaultEnv") as patch_dispose:
+            with SolverFactory("gurobi_direct") as opt:
+                opt.available()
+                opt.available()
+                PatchModel.assert_called_once_with()
+            patch_dispose.assert_not_called()
+
+
+@unittest.skipIf(not gurobipy_available, "gurobipy is not available")
+@unittest.skipIf(not single_use_license(), reason="test needs a single use license")
+class GurobiSingleUseTests(GurobiBase):
```

Comment:
I don't think these tests will ever get run with our current CI setup.

Reply:
True. I've now added additional tests in `GurobiEnvironmentTests` which cover essentially the same behaviour. I'd like to leave `GurobiSingleUseTests` in as they're useful for local development/testing.